CondFormats/HcalObjects/interface/HcalRecoParamWithPulseShapeT.h

```diff
@@ -17,7 +17,55 @@ class HcalRecoParamWithPulseShapeT {
     using PulseShapeConstView = typename PulseShapeCollection::ConstView;
     constexpr ConstView(RecoParamConstView recoView, PulseShapeConstView psView)
         : recoParamView_{recoView}, pulseShapeView_{psView} {};
-    constexpr float pulseShape(int id) const { return 0; }
+
+    constexpr float compute_pulse_shape_value(uint32_t const hashedId,
+                                              float const pulse_time,
+                                              int const sample,
+                                              int const shift) {
+      auto const recoPulseShapeId = this->recoParamView().ids()[hashedId];
+      auto const& acc25nsVec = this->pulseShapeView().acc25nsVec()[recoPulseShapeId];
+      auto const& diff25nsItvlVec = this->pulseShapeView().diff25nsItvlVec()[recoPulseShapeId];
+      auto const& accVarLenIdxMinusOneVec = this->pulseShapeView().accVarLenIdxMinusOneVec()[recoPulseShapeId];
+      auto const& diffVarItvlIdxMinusOneVec = this->pulseShapeView().diffVarItvlIdxMinusOneVec()[recoPulseShapeId];
+      auto const& accVarLenIdxZeroVec = this->pulseShapeView().accVarLenIdxZEROVec()[recoPulseShapeId];
+      auto const& diffVarItvlIdxZeroVec = this->pulseShapeView().diffVarItvlIdxZEROVec()[recoPulseShapeId];
+
+      // constants
+      constexpr float slew = 0.f;
+      constexpr auto ns_per_bx = ::hcal::constants::nsPerBX;
+
+      // FIXME: clean up all the rounding... this is coming from original cpu version
+      float const i_start_float = -::hcal::constants::iniTimeShift - pulse_time - slew > 0.f
+                                      ? 0.f
+                                      : std::abs(-::hcal::constants::iniTimeShift - pulse_time - slew) + 1.f;
+      int i_start = static_cast<int>(i_start_float);
+      float offset_start = static_cast<float>(i_start) - ::hcal::constants::iniTimeShift - pulse_time - slew;
+
+      // boundary
+      if (offset_start == 1.0f) {
+        offset_start = 0.f;
+        i_start -= 1;
+      }
+
+      int const bin_start = static_cast<int>(offset_start);
+      auto const bin_start_up = static_cast<float>(bin_start) + 0.5f;
```

Comment:
can you use `float` instead of `auto` ?

Reply:
Done in 11aa37463e89e26d17c31a9e9d76e6f799b8a298